src/control/security/auth/auth_sys_test.go

```diff
@@ -21,7 +21,7 @@
 // portions thereof marked with this legend must also reproduce the markings.
 //
 
-package auth
+package auth_test
```

Comment:
Reason for switching the package to auth_test? It seems like common practice to prefer internal tests over external tests, unless there's some compelling reason to test externally. See: https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_prefer_internal_tests_to_external_tests

Reply:
Well, I agree with Dave C. that internal tests are preferred over complicating your code or exporting things that don't need to be exported just to test them. 

In this case, these specific tests are just exercising the public API, and it increased the test coverage a little to use `security.InitDomainInfo()` vs constructing it by hand. I don't feel strongly about it though; if people would prefer that I change it back I'll do that.

Generally speaking, I have found that having at least some part of the test coverage being run from external tests is very beneficial from an API design standpoint. It can quickly highlight design flaws that internal tests might not catch because they have access to things that an API user doesn't.

Reply:
It also makes stuttering very apparent. ;) (/me looks at `auth.AuthSysRequestFromCred()`)

Reply:
auth.AuthSys isn't stuttering though. AuthSys is the name of the authentication type.

Reply:
It kinda is... You have a type in there called `Sys`, right? The package name is auth and as such it's part of the API. So `auth.SysFromToken()` would be more more idiomatic. In my opinion (which I realize is from the comfortable position of hindsight), `auth.AuthSysRequestFromCreds()` is not a great name for a public function. The name seems to imply that the caller will receive a `SysRequest` based on the supplied credentials, but that's not at all what's happening. A better name might be `auth.CredentialsFromDomainInfo()`.

Reply:
I don't feel terribly strongly about it, but it will be a pain to convert back to an internal test if we decide to do any whitebox testing of this API later. Might be worth a discussion at some point to decide when we want to test internally vs. externally in the Go code, just so we're consistent about it.

Reply:
Well, it's not a binary choice... These existing tests work fine as external tests. If more tests are added which are internal-only, it won't require these to be changed back.

Reply:
I'm just going to switch it back to an internal test. The scope really was just to fix the broken test -- changing behavior could be addressed in a future patch if desired.

Reply:
I do agree about the stutter in the naming. I believe this used to be in the 'security' package, and the name just didn't get changed when it moved over to the 'auth' package. We can address that at another time though.